src/drivers/imu/invensense/icm42670p/InvenSense_ICM42670P_registers.hpp

```diff
@@ -90,203 +82,159 @@ enum class BANK_0 : uint8_t {
 	MADDR_R           = 0x7D,
 	M_R               = 0x7E,
-
 };
 
-enum class MREG_1 : uint8_t {
-	FIFO_CONFIG5_MREG1      = 0x01,
-	INT_CONFIG0_MREG1       = 0x04,
-};
+enum class MREG1 : uint8_t {
+	FIFO_CONFIG5 = 0x01,
 
-enum class MREG_2 : uint8_t {
-	OTP_CTRL7_MREG2  = 0x06,
+	INT_CONFIG0  = 0x04,
 };
 
-enum class MREG_3 : uint8_t {
-	XA_ST_DATA_MREG3 = 0x00,
-	YA_ST_DATA_MREG3 = 0x01,
-	ZA_ST_DATA_MREG3 = 0x02,
-	XG_ST_DATA_MREG3 = 0x03,
-	YG_ST_DATA_MREG3 = 0x04,
-	ZG_ST_DATA_MREG3 = 0x05,
-};
-
-
-
 };
 
 //---------------- BANK0 Register bits
 
 // SIGNAL_PATH_RESET
 enum SIGNAL_PATH_RESET_BIT : uint8_t {
-	SOFT_RESET_DEVICE_CONFIG  = Bit4, //
-	FIFO_FLUSH                = Bit2,
+	SOFT_RESET_DEVICE_CONFIG  = Bit4, // 1: Software reset enabled
+	FIFO_FLUSH                = Bit2, // When set to 1, FIFO will get flushed
 };
 
 // INT_CONFIG
 enum INT_CONFIG_BIT : uint8_t {
-	INT1_MODE           = Bit2,
-	INT1_DRIVE_CIRCUIT  = Bit1,
-	INT1_POLARITY       = Bit0,
-};
-// GYRO_CONFIG1
-enum GYRO_CONFIG1_BIT : uint8_t {
-
-	// 2:0 GYRO_ODR
-	GYRO_UI_FILT_BW_16Hz   = Bit2 | Bit1 | Bit0,   // 111: 16Hz
-	GYRO_UI_FILT_BW_25Hz   = Bit2 | Bit1,          // 110: 25Hz
-	GYRO_UI_FILT_BW_34Hz   = Bit2 | Bit0,  	// 101: 34Hz
-	GYRO_UI_FILT_BW_53Hz   = Bit2,         	// 100: 53Hz
-	GYRO_UI_FILT_BW_73Hz   = Bit1 | Bit0,  	// 011: 73Hz
-	GYRO_UI_FILT_BW_121Hz  = Bit1,        		// 010: 121Hz
-	GYRO_UI_FILT_BW_180Hz  = Bit0,        		// 001: 180Hz
-};
-
-// ACCEL_CONFIG1
-enum ACCEL_CONFIG1_BIT : uint8_t {
-
-	// 2:0 ACCEL_ODR
-	ACCEL_UI_FILT_BW_16Hz   = Bit2 | Bit1 | Bit0,   // 111: 16Hz
-	ACCEL_UI_FILT_BW_25Hz   = Bit2 | Bit1,          // 110: 25Hz
-	ACCEL_UI_FILT_BW_34Hz   = Bit2 | Bit0,  	// 101: 34Hz
-	ACCEL_UI_FILT_BW_53Hz   = Bit2,         	// 100: 53Hz
-	ACCEL_UI_FILT_BW_73Hz   = Bit1 | Bit0,  	// 011: 73Hz
-	ACCEL_UI_FILT_BW_121Hz  = Bit1,        		// 010: 121Hz
-	ACCEL_UI_FILT_BW_180Hz  = Bit0,        		// 001: 180Hz
-};
-// FIFO_CONFIG1
-enum FIFO_CONFIG1_BIT : uint8_t {
-	// 1 FIFO_MODE
-	FIFO_MODE_STOP_ON_FULL = Bit1, // 11: STOP-on-FULL Mode
-};
-
-// INT_STATUS
-enum INT_STATUS_BIT : uint8_t {
-	RESET_DONE_INT = Bit4,
-	FIFO_THS_INT   = Bit2,
-	FIFO_FULL_INT  = Bit1,
+	INT1_MODE          = Bit2, // INT1 interrupt mode 1: Latched mode
+	INT1_DRIVE_CIRCUIT = Bit1, // INT1 drive circuit 1: Push pull
+	INT1_POLARITY      = Bit0, // INT1 interrupt polarity 0: Active low
 };
 
 // PWR_MGMT0
 enum PWR_MGMT0_BIT : uint8_t {
+	// 3:2 GYRO_MODE
 	GYRO_MODE_LOW_NOISE  = Bit3 | Bit2, // 11: Places gyroscope in Low Noise (LN) Mode
+	// 1:0 ACCEL_MODE
 	ACCEL_MODE_LOW_NOISE = Bit1 | Bit0, // 11: Places accelerometer in Low Noise (LN) Mode
 };
 
 // GYRO_CONFIG0
 enum GYRO_CONFIG0_BIT : uint8_t {
 	// 6:5 GYRO_FS_SEL
-	GYRO_FS_SEL_2000_DPS = 0,            // 0b000 = ±2000dps
-	GYRO_FS_SEL_1000_DPS = Bit5,         // 0b001 = ±1000 dps
-	GYRO_FS_SEL_500_DPS  = Bit6,         // 0b010 = ±500 dps
-	GYRO_FS_SEL_250_DPS  = Bit6 | Bit5,  // 0b011 = ±250 dps
```

Comment:
Hi @dagar .  What do you think about leaving the lower dynamic gyro and accel ranges in the enum so that if anyone wants to configure the range manually they have an easy copy paste into the register config in the header file?

Reply:
Do we have any evidence that the lower range offered any measurable improvement? At the moment I'm much more concerned with the failures that happen due to the range being insufficient (eg accel clipping).

Additionally the icm42670p supports 20 bit data output (just like the icm42688p), but we haven't implemented it. Bothering to get the full output resolution at max dynamic range is still more resolution than lower dynamic range + 16 bit raw data.

I can add 20 bit support to this driver later.